src/api/subscriptions/setSubscriptionProperty.js

```diff
@@ -2,11 +2,18 @@
 import type { ApiResponse, Auth } from '../transportTypes';
 import { apiPost } from '../apiFetch';
 
-export default async (auth: Auth, streamId: number, value: boolean): Promise<ApiResponse> =>
+export type SubscriptionProperty = 'in_home_view' | 'pin_to_top' | 'push_notifications';
+
+export default async (
+  auth: Auth,
+  streamId: number,
+  property: SubscriptionProperty,
+  value: boolean,
+): Promise<ApiResponse> =>
   apiPost(auth, 'users/me/subscriptions/properties', {
     subscription_data: JSON.stringify([
       {
-        property: 'in_home_view',
+        property,
         stream_id: streamId,
         value,
       },
```

Comment:
I do still like this consolidation! Even though it needs a bit of a wrinkle to maintain that "mute" vs. `in_home_view` facade mentioned in my previous comment.

I think a quick conditional within this function can take care of that reasonably cleanly, though. So this will take `is_muted` | `pin_to_top` | `push_notifications` as an argument, but if it's given `is_muted` it'll convert to `in_home_view` for what to actually pass to the server.